eth-riscv-runtime/src/lib.rs

```diff
@@ -73,6 +73,36 @@ pub fn revert() -> ! {
     unreachable!()
 }
 
+pub fn keccak256(offset: u64, size: u64) -> B256 {
+    let first: u64;
+    let second: u64;
+    let third: u64;
+    let fourth: u64;
+
+    unsafe {
+        asm!(
+            "ecall",
+            in("a0") offset,
+            in("a1") size,
+            lateout("a0") first,
+            lateout("a1") second,
+            lateout("a2") third,
+            lateout("a3") fourth,
+            in("t0") u32::from(Syscall::Keccak256),
+            options(nostack, preserves_flags)
```

Comment:
what does this do?

Reply:
- `lateout` - 256 bytes of keccak hash goes to 4 registers, it's like `out("a0"), out("a1") ...` etc. but marked "late" to tell that it only writes to the register after all inputs are read.

- `nostack` - tells the compiler this asm block doesn't use the stack.
- `preserves_flags` - means that `asm!` block does not modify the flags register

Saw `nostack` in other RISC-V adapters code, e.g [this](https://docs.rs/crate/riscv/latest/source/src/asm.rs#156), but it is not necessary I think. Don't have any experience with `asm!`.

Reply:
yea I feel like the options are not needed here. I think we should remove it if it works without